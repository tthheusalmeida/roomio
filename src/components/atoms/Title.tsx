import { mergeClassNames } from "@/utils/classNames";

interface TitleProps {
  children: React.ReactNode;
  className?: string;
}

export default function Title({ children, className }: TitleProps) {
  return (
    <>
      <h1 className={mergeClassNames("font-bold text-5xl my-8", className)}>
        {children}
      </h1>
    </>
  );
}
