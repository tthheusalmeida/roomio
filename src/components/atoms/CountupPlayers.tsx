import { mergeClassNames } from "@/utils/classNames";
import Title from "./Title";

interface CountupPlayersProps {
  current: number;
  max: number;
  className?: string;
}

export default function CountupPlayers({
  current,
  max,
  className,
}: CountupPlayersProps) {
  return (
    <Title
      className={mergeClassNames(
        "text-4xl font-medium flex items-center justify-center h-24",
        className
      )}
    >
      {current} / {max}
    </Title>
  );
}
