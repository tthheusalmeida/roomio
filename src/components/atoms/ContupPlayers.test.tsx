import { render, screen } from "@testing-library/react";
import CountupPlayers from "./CountupPlayers";
import Title from "./Title";

jest.mock("./Title", () => ({ children, className }: any) => (
  <div data-testid="title" className={className}>
    {children}
  </div>
));

describe("CountupPlayers component", () => {
  it("applies extra classes passed via className", () => {
    const { asFragment } = render(
      <CountupPlayers current={3} max={8} className="extra-class" />
    );

    const title = screen.getByTestId("title");
    expect(title).toHaveClass(
      "text-4xl",
      "font-medium",
      "flex",
      "items-center",
      "justify-center",
      "h-24"
    );
    expect(title).toHaveClass("extra-class");

    expect(asFragment()).toMatchSnapshot();
  });
});
