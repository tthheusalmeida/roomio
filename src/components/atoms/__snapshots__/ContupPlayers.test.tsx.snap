// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CountupPlayers component applies extra classes passed via className 1`] = `
<DocumentFragment>
  <div
    class="text-4xl font-medium flex items-center justify-center h-24 extra-class"
    data-testid="title"
  >
    3 / 8
  </div>
</DocumentFragment>
`;
